UnitySDK/Assets/ML-Agents/Scripts/RewardProvider/BaseRewardProviderComponent.cs

```diff
@@ -0,0 +1,15 @@
+using System;
+using UnityEngine;
+
+namespace MLAgents.RewardProvider
+{
+    public class BaseRewardProviderComponent: MonoBehaviour
```

Comment:
There is a lot about the Reward providers as component in this PR, can you write a few lines on how it is supposed to work? For example, I do not think all reward providers should be wrapped into a BaseRewardProviderComponent to be usable by an agent. What do you think?

Reply:
I think if we want users to be able to drag reward providers onto agents in the editor then we need them to have component counterparts.  Otherwise they will be instantiating them on their own.  I think part of the usefulness of this feature is the fact that you can drag and drop them onto your agent in the editor.

Reply:
There is no longer a BaseRewardProviderComponent.  Only a LowLevelRewardProviderComponent which is created by the Agent if one is not present.  This wraps the LowLevelRewardProvider and continues to allow users to use the `AddReward`(etc.) APIs on Agent.